src/Data/Vec/Properties.agda

```diff
@@ -1108,20 +1123,20 @@ remove-punchOut (x ∷ y ∷ xs) {suc i} {suc j} i≢j =
   remove-punchOut (y ∷ xs) (i≢j ∘ cong suc)
 
 ------------------------------------------------------------------------
--- remove
+-- insertAt and removeAt
 
-remove-insert : ∀ (xs : Vec A n) (i : Fin (suc n)) (v : A) →
-                remove (insert xs i v) i ≡ xs
-remove-insert xs           zero           v = refl
-remove-insert (x ∷ xs)     (suc zero)     v = refl
-remove-insert (x ∷ y ∷ xs) (suc (suc i))  v =
-  cong (x ∷_) (remove-insert (y ∷ xs) (suc i) v)
+removeAt-insertAt : ∀ (xs : Vec A n) (i : Fin (suc n)) (v : A) →
+                    removeAt (insertAt xs i v) i ≡ xs
+removeAt-insertAt xs           zero           v = refl
+removeAt-insertAt (x ∷ xs)     (suc zero)     v = refl
+removeAt-insertAt (x ∷ y ∷ xs) (suc (suc i))  v =
+  cong (x ∷_) (removeAt-insertAt (y ∷ xs) (suc i) v)
 
-insert-remove : ∀ (xs : Vec A (suc n)) (i : Fin (suc n)) →
-                insert (remove xs i) i (lookup xs i) ≡ xs
-insert-remove (x ∷ xs)     zero     = refl
-insert-remove (x ∷ y ∷ xs) (suc i)  =
-  cong (x ∷_) (insert-remove (y ∷ xs) i)
+insertAt-removeAt : ∀ (xs : Vec A (suc n)) (i : Fin (suc n)) →
+                    insertAt (removeAt xs i) i (lookup xs i) ≡ xs
+insertAt-removeAt (x ∷ xs)     zero     = refl
+insertAt-removeAt (x ∷ y ∷ xs) (suc i)  =
```

Comment:
Use an `@`-pattern: `xs@(_ ∷ _)` and then you don't need the redundant repetition of the cons on the RHS; simply `xs` suffices (think about the compiler, and the poor allocator in the run-time, if this were a vanilla functional language: you've destructed a cons cell, only to then put it back together again... ;-))